### PR TITLE
libaosd: init at 0.2.7-9-g177589f

### PR DESCRIPTION
**Pull request by unode**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by c0bw3b** [comment]
Forgive my question, but since this library is unmaintained and already legacy (last commit in 2011), is it really worth adding it to \<nixpkgs\> now?

If you have a private use for it, you could build it locally and push it on [Cachix](https://cachix.org/).
Or maybe you intended to package something later that would depend on this?

I'm genuinely curious here, not trying to imply this should be turned down just because it's old
(although _it is quite old_ :panda_face: )

**Comment by unode** [comment]
Hi Renaud, Yes I am [using it](https://github.com/unode/MouseWarp) and since it worked out of the box with minimal effort I found no reason not to include it in nixpkgs. It's present in [other distros](https://repology.org/metapackage/libaosd/versions) and some software builds on it.

I don't think age should be a major blocker as long as it works. Which is to say that if something does break and is not easily fixed, it would be a candidate for removal. For now it's all in the green.

I also saw that there are [aosd haskell bindings in nixpkgs](https://nixos.org/nixos/packages.html#libaosd) but I didn't find a standalone `libaosd`.

File: pkgs/development/libraries/libaosd/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  name = "libaosd-${version}";
+
+  src = fetchFromGitHub {
+    owner  = "atheme-legacy";
+    repo   = "libaosd";
+    rev    = "${version}";
+    sha256 = "1cn7k0n74p6jp25kxwcyblhmbdvgw3mikvj0m2jh4c6xccfrgb9a";
+  };
+
+  buildInputs = [autoconf automake pkgconfig cairo pango libX11 libXcomposite];
```
Comment (c0bw3b) [inline pkgs/development/libraries/libaosd/default.nix]:
```suggestion
  nativeBuildInputs = [ autoconf automake pkgconfig ];
  buildInputs = [ cairo pango libX11 libXcomposite ];
  enableParallelBuilding = true;
```

File: pkgs/development/libraries/libaosd/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  prePatch = ''
```
Comment (c0bw3b) [inline pkgs/development/libraries/libaosd/default.nix]:
```suggestion
  preConfigure = ''
```
would be more sensible

**Review by c0bw3b — CHANGES_REQUESTED** [review]

**Comment by unode** [comment]
Changes implemented

**Review by c0bw3b — APPROVED** [review]

**Pull request merged**